src/CLI/clitree/cli-xml/nat.xml

```diff
@@ -167,7 +167,14 @@ limitations under the License.
              optional="true"
              >
          </PARAM>
-         <ACTION builtin="clish_nop"></ACTION>
+
+         <ACTION> 
+             if test "${global-port-range}" = ""; then&#xA;
+                 python $SONIC_CLI_ROOT/sonic-cli-nat.py patch_openconfig_nat_nat_instances_instance_nat_pool_nat_pool_entry_config ${pool-name} ${global-ip-range}&#xA; 
+             else&#xA;
+                 python $SONIC_CLI_ROOT/sonic-cli-nat.py patch_openconfig_nat_nat_instances_instance_nat_pool_nat_pool_entry_config ${pool-name} ${global-ip-range} ${global-port-range}&#xA;
```

Comment:
Pass both the parameters always and validate in the script side.
I know existing code is  doing it this way but it would be cleaner handling it in the script.